src/test/java/com/google/sps/AuthCheckerTest.java

```diff
@@ -0,0 +1,47 @@
+package com.google.sps;
+
+import javax.servlet.http.HttpServletResponse;
+
+import com.google.appengine.api.users.User;
+import com.google.appengine.api.users.UserService;
+import com.google.sps.util.AuthChecker;
+
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import static org.mockito.Mockito.*;
+
+import java.io.IOException;
+
+@RunWith(JUnit4.class)
+public class AuthCheckerTest {
+  private HttpServletResponse response;
+  private UserService userServiceNotLoggedIn;
+  private UserService userServiceLoggedIn;
+
+  @Before
+  public void setUp() throws IOException {
+    response = mock(HttpServletResponse.class);
+    userServiceNotLoggedIn = mock(UserService.class);
+    userServiceLoggedIn = mock(UserService.class);
+
+    when(userServiceNotLoggedIn.isUserLoggedIn()).thenReturn(false);
+    when(userServiceLoggedIn.isUserLoggedIn()).thenReturn(true);
+    when(userServiceLoggedIn.getCurrentUser()).thenReturn(new User("peter@google.com", "overnightly.com"));
+  }
+
+  @Test
+  public void testUnauthenticatedUser() {
+    String email = AuthChecker.getUserEmail(response, userServiceNotLoggedIn);
+    Assert.assertEquals(null, email);
```

Comment:
This can be changed to assertNull(email), and it should do the same thing. Small though so it doesn't' really matter.

Reply:
Done. I actually didn't know about that method, thanks for pointing it out!

Reply:
No problem! Good PR

Reply:
Good discussion. Yeah, assertNull is better than assertEquals because it is easier to read. Google actually has a library for more fluent assertion for everything: https://github.com/google/truth